department-of-veterans-affairs/abd-vro
# MCP-2574 Continue Processing when BIP CE is down

## What was the problem?
When BIP CE API is down, previously, processing would just stop and fail. 

Associated tickets or Slack threads:
- [MCP-2574](https://amida.atlassian.net/browse/MCP-2574)

## How does this fix it?[
Now we notify slack, and continue processing successfully

## How to test this PR
Again, another doozy to test.

Set up your environment with mocks standing up, and whatever slack config you want to test. I use an end to end test config but override the application yml to have the non prod slack channel active.

Stand docker up, stop mock BIP CE container. *do not* take down BIP Claim API. Only CE.

Send MAS an automated processing claim for a collection ID setup for ordering an exam. Id 377 and 378 work.
You can use VROV2Tests.java for this, but you'll have to comment out this line in setup

    // restTemplate.delete(RECEIVED_FILES_URL + fileNumber); (line 288)

Then, run the test from there (or run it via another method).

The logs will show an error with BIP exception which will get handled, and a slack message with the appropriate info will get called. You'll also see the logs continue on and update the BIP Claim API.

Attached screenshot of what you get message wise in slack. 
<img width="467" alt="Screen Shot 2023-03-24 at 11 54 00 PM" src="https://user-images.githubusercontent.com/57402207/227692644-6344fbc3-6b7c-4765-8f68-8a719c465079.png">
